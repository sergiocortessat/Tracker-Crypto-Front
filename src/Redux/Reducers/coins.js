const initialState = {
  coins: [],
};

const coinReducer = (state = initialState, action) => {
  switch (action.type) {
    case 'ALL_COINS':
      return {
        ...state,
        coins: action.payload,
      };
    default:
      return state;
  }
};

export default coinReducer;
