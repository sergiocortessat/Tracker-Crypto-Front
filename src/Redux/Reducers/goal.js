const initialState = {
  goals: [],
};

const goalReducer = (state = initialState, action) => {
  switch (action.type) {
    case 'ALL_GOALS':
      return {
        ...state,
        goal: action.payload,
      };
    default:
      return state;
  }
};

export default goalReducer;
