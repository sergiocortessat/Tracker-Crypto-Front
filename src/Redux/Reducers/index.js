import { combineReducers } from 'redux';
import Coins from './coins';
import User from './user';

const reducers = combineReducers({
  coins: Coins,
  user: User,
});

export default reducers;
