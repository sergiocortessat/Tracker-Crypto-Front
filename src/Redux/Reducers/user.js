const initialState = {
  user: [],
};

const userReducer = (state = initialState, action) => {
  switch (action.type) {
    case 'CURRENT_USER':
      return {
        ...state,
        user: action.payload,
      };
    default:
      return state;
  }
};

export default userReducer;
