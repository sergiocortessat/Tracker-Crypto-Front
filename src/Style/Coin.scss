$grey: #67899c;
$green: #addc91;
$blue: #62b5e5;

.coins-container {
  background-color: lightgrey;
  padding: 15px;
  display: flex;
  align-items: center;
  flex: 1;
}

.all-coins {
  display: flex;
  flex-wrap: wrap;
  width: 90%;
  margin: 0 auto;
  align-items: center;

  a {
    width: 100%;
    margin-right: 10px;
    margin-bottom: 10px;
    background: rgba($color: white, $alpha: 0.9);
    border-radius: 5px;
  }

  .main-coin {
    display: flex;
    width: 50%;
    // margin-bottom: 40px;
  }
}

.coin-list-link {
  text-decoration: none;
}

.coin {
  display: flex;
  flex-direction: row;
  align-items: center;
  height: 100px;

  a {
    width: 100%;
  }

  .crypto-info {
    text-align: center;
    padding: 5px 3px;

    h3 {
      color: $grey;
    }

    h4 {
      color: black;
    }
  }

  .crypto-icon {
    padding: 10px 10px;

    img {
      border-radius: 50%;
      width: 40px;
      height: 48px;
    }
  }
}

@media screen and (min-width: 768px) {
  .all-coins {
    width: 800px;
  }
}
