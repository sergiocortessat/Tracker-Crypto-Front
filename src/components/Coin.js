/* eslint-disable array-callback-return */
/* eslint-disable react/prop-types */
/* eslint-disable no-unused-vars */
import { useAuth0 } from '@auth0/auth0-react';
import { set } from 'lodash';
import React, { useEffect, useState } from 'react';
import { getGoal } from '../API/API';

const Coin = ({ coins, setSum }) => {
  const { user, isAuthenticated, getAccessTokenSilently } = useAuth0();
  const [goal, setGoal] = useState([]);

  useEffect(() => {
    getGoal(user.sub).then((res) => {
      const { goal, measurements } = res.filter((item) => item.coin_id === coins.id)[0];
      setGoal(goal);
      setSum((prev) => prev + goal);
      // console.log(sum);
    });
  }, []);

  return (
    <div key={coins.id} className="coin">
      <div className="crypto-icon">
        <img src={coins.picture} alt={coins.name} />
      </div>
      <div className="crypto-info">
        <p>{coins.name}</p>
        <p>
          Goal:
          {' '}
          {goal && goal}
        </p>
      </div>
    </div>
  );
};

export default Coin;
