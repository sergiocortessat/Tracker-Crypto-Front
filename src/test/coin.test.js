import { Provider } from 'react-redux';
import { createStore } from 'redux';
import renderer from 'react-test-renderer';
import Reducers from '../Redux/Reducers';
import Coin from '../components/Coin';

const store = createStore(Reducers);

it('renders correctly', () => {
  const addMeasurement = renderer
    .create(
      <Provider store={store}>
        <Coin />
      </Provider>,
    )
    .toJSON();
  expect(addMeasurement).toMatchSnapshot();
});
