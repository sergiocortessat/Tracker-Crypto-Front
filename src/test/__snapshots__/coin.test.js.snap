// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="coin"
>
  <div
    className="crypto-icon"
  >
    <img
      alt=""
      src=""
    />
  </div>
  <div
    className="crypto-info"
  >
    <h3>
      
    </h3>
    <h4>
      Goal:
       
    </h4>
  </div>
</div>
`;
